.github/workflows/pr.yml

```diff
@@ -2,26 +2,41 @@ name: PR Workflow
 
 on:
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 jobs:
   lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '15.x'
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
 
+      - name: Install
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Lint
+        run: yarn lint
+
+  prod-install:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Use Node.js
-      uses: actions/setup-node@v2
-      with:
-        node-version: '15.x'
-    
-    - name: Install dependencies
-      run: yarn install
-    
-    - name: Bootstrap
-      run: yarn bootstrap  
-
-    - name: Lint
-      run: yarn lint
-      
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '15.x'
+
+      - name: Install dependencies
+        run: yarn install --prod
```

Comment:
I think we can completely remove this bit. I think I didnt explain well what is the goal of this issue apologies. Lets forget what I told you about the production build. I would say its premature optimisation at this point

Reply:
> Let me know if that makes sense the way I am describing it

It makes sense. I added the typechking and the code runs without errors locally.

Nevertheless the Github Action is throwing this error again: `error TS2307: Cannot find module '@small-ads/ui' or its corresponding type declarations`. I am trying to solve it.